### ISSUE TITLE
Sample for Built-in module *iconv* in documentation has error?

### ISSUE THREAD

**Issue by zaraysky** [tarantool/doc#745]
Documentation https://tarantool.io/en/doc/1.10/reference/reference_lua/iconv/ has sample
```
string.hex('Д')
converter = require('iconv').new('UTF16BE', 'UTF8')
utf16_string = converter('Д')
string.hex(utf16_string)
```
Parameter `UTF16BE ` for iconv.new() is wrong here. Must be `UTF-16`
```
string.hex('Д')
converter = require('iconv').new('UTF-16', 'UTF8')
utf16_string = converter('Д')
string.hex(utf16_string)
```

**Comment by Totktonada** [comment]
It depends of iconv implementation under hood of iconv lua module. Maybe a note about that need to be added. See also https://github.com/tarantool/tarantool/issues/3791

**Comment by pgulutzan** [comment]
Does anyone know whether UTF16BE causes failure with any Linux, or with Mac? Does anyone know whether UTF-16 always works with all Linux, and with Mac, and with FreeBSD? Will the fix for issue#3791 make this problem disappear?

**Comment by pgulutzan** [comment]
I did not get an answer to the above questions. I tried to research what iconv always accepts and what the variations are, but did not find anything clear. So I accepted Toktonada's suggestion and added a note.
